hera_cal/tests/test_lstbin.py

```diff
@@ -268,6 +317,132 @@ def test_lst_bin_files(self):
+        reds_data_missing[1] = reds_data_missing[1][2:]
+        bls_missing_first_group = []
+        for grp in reds_data_missing:
+            for bl in grp:
+                bls_missing_first_group.append(bl)
+        # replaced redundantly averaged files in second
+        # and third nights with data where
+        # first redundant group is missing two baselines.
+        for flist in self.data_files[1:]:
+            for fstr in flist:
+                hd = io.HERAData(fstr)
+                hd.read(bls=bls_missing_first_group)
+                utils.red_average(hd, inplace=True, reds=reds_data)
+                out_path = os.path.join(tmp_path, fstr.split('/')[-1].replace('.uvh5', '.red_average.uvh5'))
+                hd.write_uvh5(out_path, clobber=True)
+
+        lstbin.lst_bin_files(redundantly_averaged_filepaths, outdir=tmp_path, overwrite=True, median=False,
+                             verbose=False, file_ext=file_ext, ignore_flags=False, average_redundant_baselines=True)
+
+        uv4 = io.HERAData(output_lst_file)
+        d4, f4, n4 = uv4.read()
+        # assert that all nsamples and data of lstbinned and red averaged equal to red average then lst bin.
+        for k in d2:
+            if not(k[:2] in reds_data[1] or k[:2][::-1] in reds_data[1]):
+                assert np.all(np.isclose(n4[k], n2[k]))
+                assert np.all(np.isclose(d4[k], d2[k]))
+                assert np.all(np.isclose(f4[k], f2[k]))
+        # remove all but a single antenna from data on the first night.
+        for fnum, flist in enumerate(self.data_files):
+            for fstr in flist:
+                hd = UVData()
+                hd.read(fstr)
+                nants = len(hd.antenna_numbers)
+                # on first night, remove all but two antennas.
+                if fnum < 2:
+                    hd.select(antenna_nums=np.unique(hd.ant_1_array)[:2], keep_all_metadata=False)
+                hd.write_uvh5(os.path.join(tmp_path, 'temp.uvh5'), clobber=True)
+                hd = io.HERAData(os.path.join(tmp_path, 'temp.uvh5'))
+                hd.read()
+                if fnum < 2:
+                    assert len(hd.antpos) < nants
+                utils.red_average(hd, inplace=True, reds=reds_data)
+                out_path = os.path.join(tmp_path, fstr.split('/')[-1].replace('.uvh5', '.red_average.uvh5'))
+                hd.write_uvh5(out_path, clobber=True)
+
+        lstbin.lst_bin_files(redundantly_averaged_filepaths, outdir=tmp_path, overwrite=True, median=False,
+                             verbose=False, file_ext=file_ext, ignore_flags=False, average_redundant_baselines=True)
+        # assert that the keys when only two antennas are present on the first night data
+        # are identical to the data set when all antennas are present on the first night data
+        uv5 = io.HERAData(output_lst_file)
+        d5, f5, n5 = uv5.read()
+        assert sorted(list(d5.keys())) == sorted(list(d4.keys()))
+
```

Comment:
gen_bldicts should probably have its own broken-out unit test. How hard would that be to do?

Reply:
I've broken out a short unittest.